models.py

```diff
@@ -15,104 +15,134 @@ class EVC:
     """Class that represents a E-Line Virtual Connection."""
 
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, name, uni_a, uni_z, start_date=None, end_date=None,
+                 bandwidth=None, primary_links=None, backup_links=None,
+                 dynamic_backup_path=None, creation_time=None):
         """Create an EVC instance with the provided parameters.
 
         Do some basic validations to attributes.
         """
-        self.validate(*args, **kwargs)
-        self.fill_attributes(*args,**kwargs)
+        self.validate(name, uni_a, uni_z)
+
+        self._id = uuid4().hex
+        self.uni_a = uni_a
+        self.uni_z = uni_z
+        self.name = name
+        self.start_date = start_date if start_date else now()
+        self.end_date = end_date
+        # Bandwidth profile
+        self.bandwidth = bandwidth
+        self.primary_links = primary_links
+        self.backup_links = backup_links
+        self.dynamic_backup_path = dynamic_backup_path
+        # dict with the user original request (input)
+        self._requested = {}
+        # circuit being used at the moment if this is an active circuit
+        self.current_path = []
+        # primary circuit offered to user IF one or more links were provided in
+        # the request
+        self.primary_path = []
+        # backup circuit offered to the user IF one or more links were provided
+        # in the request
+        self.backup_path = []
+        # datetime of user request for a EVC (or datetime when object was
+        # created)
+        self.request_time = now()
+        # datetime when the circuit should be activated. now() || schedule()
+        self.creation_time =  creation_time or now()
+        self.owner = None
+        # Operational State
+        self.active = False
+        # Administrative State
+        self.enabled = False
+        # Service level provided in the request. "Gold", "Silver", ...
+        self.priority = 0
+
+    def validate(self, name=None, uni_a=None, uni_z=None):
+        """Validate the EVC arguments.
 
-    def fill_attributes(self, *args, **kwargs):
-        """Fill all attributes given in the EVC object."""
-        for attribute, default in self.default_attributes.items():
-            value = kwargs.get(attribute, default)
-
-            if attribute == '_id':
-                value = kwargs.get('id', default)
+        Raises:
+            TypeError: Raises an error with the error message, when the params
+                       are invalid.
+        """
+        # Verify required attributes
```

Comment:
I think ValueError is the right exception here, not TypeError.

Also, "Raises an error with the error message" is redundant, because it's true for any exception. Just say "ValueError: raised when object attributes are invalid" or something like that.

The commentary that it "Verifies required attributes" is redundant too.